docs/development/go/websocket/index.md

```diff
@@ -0,0 +1,651 @@
+[INFO  websocat::stdio_peer] get_stdio_peer (async)
+[INFO  websocat::stdio_peer] Setting stdin to nonblocking mode
+[INFO  websocat::stdio_peer] Installing signal handler
+[INFO  websocat::ws_client_peer] get_ws_client_peer
+[INFO  websocat::ws_client_peer] Connected to ws
+Hello!
+Hello!
+[INFO  websocat::sessionserve] Forward finished
+[INFO  websocat::sessionserve] Forward shutdown finished
+[INFO  websocat::sessionserve] Reverse finished
+[INFO  websocat::sessionserve] Reverse shutdown finished
+[INFO  websocat::sessionserve] Finished
+[INFO  websocat::stdio_peer] Restoring blocking status for stdin
+[INFO  websocat::stdio_peer] Restoring blocking status for stdin
+{{< /output >}}
+
+For these two `websocat` interactions, the WebSocket server generated the following output:
+
+{{< output >}}
+2020/07/06 19:00:48 Connection from: localhost:1234
+2020/07/06 19:00:58 Received: Hello there from websocat!
+2020/07/06 19:01:02 Received: Bye!
+2020/07/06 19:01:03 From localhost:1234 read websocket: close 1005 (no status)
+2020/07/06 19:02:23 Received: Hello!
+2020/07/06 19:02:23 From localhost:1234 read websocket: close 1005 (no status)
+{{< /output >}}
+
+## Developing WebSocket clients in Go
+
+In this section of the guide you will learn how to develop WebSocket clients using two different Go packages.
+
+### Using `gorilla/websocket`
+
+The subject of this section is the development of WebSocket clients using the functionality offered by `gorilla/websocket`.
+
+#### The Go code of the Client
+
+The present WebSocket client requires two command line arguments, which are the hostname of the WebSocket server with the port number and the endpoint. The Go code of `WSclient.go` is the following:
+
+{{< file "./WSclient.go" go >}}
+package main
+
+import (
+  "bufio"
+  "fmt"
+  "log"
+  "net/url"
+  "os"
+  "os/signal"
+  "syscall"
+  "time"
+
+  "github.com/gorilla/websocket"
+)
+
+var SERVER = "ws://localhost:1234/"
+var PATH = ""
+var TIMESWAIT = 0
+var TIMESWAITMAX = 5
+
+func getInput(input chan string) {
+  in := bufio.NewReader(os.Stdin)
```

Comment:
The bufio.Reader and any data buffered by it is discarded on return from the function.  

The code functions as expected when a user is entering data from a terminal, but fails when the input is from a file.  In the file scenario, the call to in.ReadString('\n') will read more than one line of text from stdin and this extra text is discarded.

This is a common error in the author's code that applies to reading stdin and network connections.  See related issue https://github.com/linode/docs/issues/3388.

Create a single bufio.Reader for stdin and use it for the lifetime of the application.